src/app.ts

```diff
@@ -281,12 +297,16 @@ function appInit(gopt: AppOpt = {}): App {
 
 	app.canvas.addEventListener("touchend", (e) => {
 		if (!gopt.touchToMouse) return;
-		app.mouseState = "released";
+		app.mouseStates["left"] = "released";
 	});
 
 	app.canvas.addEventListener("touchcancel", (e) => {
 		if (!gopt.touchToMouse) return;
-		app.mouseState = "released";
+		app.mouseStates["left"] = "released";
+	});
+
+	app.canvas.addEventListener("contextmenu", function (e) {
+		e.preventDefault();
 	});
```

Comment:
kinda unrelated, but context menu doesn't make much sense in a canvas, and we support customizing right mouse button now